components/eventing-controller/pkg/handlers/beb.go

```diff
@@ -191,6 +202,29 @@ func (b *Beb) deleteCreateAndHashSubscription(subscription *types.Subscription,
 	return bebSubscription, newEmsHash, nil
 }
 
+func (b *Beb) createAndGetSubscription(subscription *types.Subscription, cleaner eventtype.Cleaner, log *zap.SugaredLogger) (*types.Subscription, error) {
+	// clean the application name segment in the subscription event-types from none-alphanumeric characters
+	if err := cleanEventTypes(subscription, cleaner); err != nil {
+		log.Errorw("clean application name in the subscription event-types failed", "error", err)
+		return nil, err
+	}
+
+	// create a new EMS subscription
+	if err := b.createSubscription(subscription, log); err != nil {
+		log.Errorw("create BEB subscription failed", "bebSubscriptionName", subscription.Name, "error", err)
+		return nil, err
+	}
+
+	// get the new EMS subscription
+	bebSubscription, err := b.getSubscription(subscription.Name)
```

Comment:
what does it mean if we fail here? we created a subscription successfully, but we cannot retrieve it. does it mean we just have to wait a bit, or is something really wrong on the backend side?

Reply:
Something is wrong on BEB side. We will try to redo the action at the next reconcile call.